presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveArrayFunctionQueries.java

```diff
@@ -45,4 +45,54 @@ public void testShuffle()
         this.assertQuerySucceeds("SELECT shuffle(array_sort(quantities)) FROM orders_ex");
         this.assertQuery("SELECT array_sort(shuffle(quantities)) FROM orders_ex");
     }
+
+    @Test
+    public void testArrayMatch()
+    {
+        final String errorMessage = ".*division by zero presto.default.divide.*";
+        final String expectTrue = "SELECT true";
+        final String expectFalse = "SELECT false";
+        final String expectNull = "SELECT null";
+        // test all_match
+        this.assertQuerySucceeds("SELECT all_match(quantities, x -> ((10 / x) > 2)) FROM orders_ex");
+        this.assertQueryFails("SELECT all_match(array[1, 2, 0], x -> ((10 / x) > 2))", errorMessage);
+        this.assertQuery("SELECT all_match(array[1, 5, 0], x -> ((10 / x) > 2))", expectFalse);
```

Comment:
Why do we need to use expectTrue/False/Null queries? Why can't we just do this:

```
assertQuery("SELECT all_match(array[1, 5, 0], x -> ((10 / x) > 2))");
```

Reply:
Got it, fixed. It turns out that `assertQuery` has an `ExpectedQueryRunner` to get the expected results:)